src/toolchain/toolchain.ts

```diff
@@ -156,4 +162,33 @@ export class SwiftToolchain {
             throw Error("Cannot find swift executable.");
         }
     }
+
+    /**
+     * @returns if the default Swift driver is the new driver
+     */
+    private static async judgeNewDriver(toolchainPath: string): Promise<boolean> {
+        const toolDirectory = path.join(toolchainPath, "usr", "bin");
+        // judge from environment variable
+        if (process.env.SWIFT_USE_NEW_DRIVER) {
+            return true;
+        }
+        if (process.env.SWIFT_USE_OLD_DRIVER) {
+            return false;
+        }
+        // judge from tool existence
+        if (await pathExists(toolDirectory, getExecutableName("swift-driver"))) {
+            return true;
+        }
+        if (await !pathExists(toolDirectory, getExecutableName("swift-frontend"))) {
+            return false;
+        }
+        // check if swift is replaced by the new driver
+        const swiftDriverBuffer = await fs.readFile(
+            path.join(toolDirectory, getExecutableName("swift"))
+        );
+        const swiftFrontendBuffer = await fs.readFile(
+            path.join(toolDirectory, getExecutableName("swift-frontend"))
+        );
+        return !swiftDriverBuffer.equals(swiftFrontendBuffer);
```

Comment:
Instead of comparing buffers, could you not check the symbolic links point to the same source.

Reply:
Swift tool binaries are _copied_ instead of _linked_ on Windows most of the time, so comparing the file itself is the correct way.

Reply:
I added symlink comparison before buffer. Mark this as resolved